x-pack/plugins/observability_ai_assistant/public/components/insight/insight.tsx

```diff
@@ -142,18 +213,87 @@ export function Insight({ messages, title, dataTestSubj }: InsightProps) {
     [service]
   );
 
+  const handleSend = (newPrompt: string) => {
+    const clonedMessages = cloneDeep(messages);
+    const message = last(clonedMessages.filter((msg) => msg.message.role === MessageRole.User));
```

Comment:
nit: perhaps a utility function to get the last message of a particular type since it's a common operation (I see it three times in this file):
```ts
function getLastMessageOfType(messages: Message[], role: MessageRole) {
  return last(messages.filter((msg) => msg.message.role === MessageRole.User));
}
```


```suggestion
    const userMessage = getLastMessageOfType(clonedMessages, MessageRole.User);
```

Reply:
updated in https://github.com/elastic/kibana/pull/176086/commits/fa7d3601e5da7a355db4969da56fa305708c5987